testing/integration/upgrade_test.go

```diff
@@ -304,40 +285,103 @@ func (s *StandaloneUpgradeTestSuite) TestUpgradeStandaloneElasticAgentToSnapshot
 		buildFragments[1],
 	)
 
-	s.T().Logf("Upgrading to version %q", upgradeInputVersion)
+	testUpgrade(ctx, t, agentFixture, upgradeInputVersion.String(), expectedAgentHashAfterUpgrade)
+
+}
+
+func getUpgradableAndLatestVersions(ctx context.Context, t *testing.T) (upgradableVersions []string, latestVersion string) {
+	t.Helper()
+	aac := tools.NewArtifactAPIClient()
+	vList, err := aac.GetVersions(ctx)
+	require.NoError(t, err, "error retrieving versions from Artifact API")
+	require.NotNil(t, vList)
```

Comment:
This would not only check that `vList` is not `nil` but also that it doesn't have a length of 0.
```suggestion
	require.NotEmpty(t, vList)
```